From aec48eef1619cd95e0286149e2dedb2dc3df1f05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Tue, 3 Nov 2020 16:51:34 +0200
Subject: [PATCH 037/171] qglobal: Only define QT_ENSURE_STACK_ALIGNED_FOR_SSE
 for i386

This define is only supposed to be used for i386, but was set for
any 32 bit mingw architecture (which also covers armv7).

Change-Id: Iedc057dfc493015e8339db837dbe20a57c2b2367
Reviewed-by: Friedemann Kleint <Friedemann.Kleint@qt.io>
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 317145af84c055cf019c84b9090f53bd8c61c5da)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/global/qglobal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/global/qglobal.h b/src/corelib/global/qglobal.h
index 7aa856a65d..a024e69751 100644
--- a/src/corelib/global/qglobal.h
+++ b/src/corelib/global/qglobal.h
@@ -632,7 +632,7 @@ using qsizetype = QIntegerForSizeof<std::size_t>::Signed;
 //defines the type for the WNDPROC on windows
 //the alignment needs to be forced for sse2 to not crash with mingw
 #if defined(Q_OS_WIN)
-#  if defined(Q_CC_MINGW) && !defined(Q_OS_WIN64)
+#  if defined(Q_CC_MINGW) && defined(Q_PROCESSOR_X86_32)
 #    define QT_ENSURE_STACK_ALIGNED_FOR_SSE __attribute__ ((force_align_arg_pointer))
 #  else
 #    define QT_ENSURE_STACK_ALIGNED_FOR_SSE
-- 
2.31.1

